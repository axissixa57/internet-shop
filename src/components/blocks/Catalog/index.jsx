import React from 'react'
import { Row, Col, Card, Rate, Tag, Button } from 'antd'
import styled from 'styled-components'

const SectionCatalog = styled.div`
  padding: 1rem;
`

const RowComponent = styled(Row)`
  display: flex;
  flex-flow: row wrap;

  .ant-card-cover {
    img {
      padding: 0.5rem;
    }
  }

  .ant-card-body {
    padding: 1rem;

    div:first-child > span {
      font-weight: 600;
      font-size: 1.5rem;
    }
  }

  .ant-card-meta-description {
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
`

const Tags = styled.div`
  margin-top: 0.5rem;
`

const PriceHeader = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 0.5rem;
  align-items: center;
`

const Icons = styled.div`
  button {
    font-size: 1.4rem;
  }
`

const { Meta } = Card

const Catalog = () => (
  <SectionCatalog>
    <RowComponent>
      <Col
        xs={24} sm={24} md={12} lg={12}
        xl={6}
      >
        <Card
          hoverable
          style={{ width: 240, marginBottom: '1rem' }}
          cover={
            <img
              alt="example"
              src="https://gloimg.samcdn.com/S/pdm-product-pic/Clothing/2016/06/03/source-img/20160603190149_72043.jpg" />
          }
        >
          <PriceHeader>
            <span>5$</span>
            <Icons>
              <Button type="link" shape="circle" icon="heart" />
              <Button type="link" shape="circle" icon="shopping-cart" />
            </Icons>
          </PriceHeader>
          <Meta
            title="Polo red"
            description="uspendisse nisi erat, elementum nec mollis ut, dignissim" />
          <Rate disabled allowHalf defaultValue={3.5} />
          <Tags>
            <Tag color="red" key="red">
              {'red'.toUpperCase()}
            </Tag>
          </Tags>
        </Card>
      </Col>
      <Col
        xs={24} sm={24} md={12} lg={12}
        xl={6}
      >
        <Card
          hoverable
          style={{ width: 240, marginBottom: '1rem' }}
          cover={
            <img
              alt="example"
              src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png" />
          }
        >
          <Meta title="Europe Street beat" description="www.instagram.com" />
        </Card>
      </Col>
      <Col
        xs={24} sm={24} md={12} lg={12}
        xl={6}
      >
        <Card
          hoverable
          style={{ width: 240, marginBottom: '1rem' }}
          cover={
            <img
              alt="example"
              src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png" />
          }
        >
          <Meta title="Europe Street beat" description="www.instagram.com" />
        </Card>
      </Col>
      <Col
        xs={24} sm={24} md={12} lg={12}
        xl={6}
      >
        <Card
          hoverable
          style={{ width: 240, marginBottom: '1rem' }}
          cover={
            <img
              alt="example"
              src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png" />
          }
        >
          <Meta title="Europe Street beat" description="www.instagram.com" />
        </Card>
      </Col>
    </RowComponent>
  </SectionCatalog>
)

export default Catalog
