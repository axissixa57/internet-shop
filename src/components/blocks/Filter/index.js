export { default as Filter } from './container'
