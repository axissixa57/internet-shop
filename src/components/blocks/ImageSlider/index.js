export { default as ImageSlider } from './component'
