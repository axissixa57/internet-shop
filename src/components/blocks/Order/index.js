export { default as Order } from './container'
