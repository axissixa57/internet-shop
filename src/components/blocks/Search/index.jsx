import React from 'react'
import { Input } from 'antd'

const Search = () => {
  return (
    <div className="search">
      <Input.Search placeholder="Поиск" />
    </div>
  )
}

export default Search
