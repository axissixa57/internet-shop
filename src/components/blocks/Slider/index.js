export { default as Slider } from './container'
