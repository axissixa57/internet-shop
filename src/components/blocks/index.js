export { Catalog } from './Catalog'
export { Search } from './Search'
export { Userbar } from './Userbar'
export { Header } from './Header'
export { Order } from './Order'
export { Filter } from './Filter'
