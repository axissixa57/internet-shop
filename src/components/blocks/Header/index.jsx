import React from 'react'
import { Header } from './styles'

import { Search, UserBar } from '@/components/blocks/'

const HeaderComponent = () => {
  return (
    <Header
      title="TestShop"
      subTitle={
        <>
          <Search />
          <UserBar />
        </>
      } />
  )
}

export default HeaderComponent
