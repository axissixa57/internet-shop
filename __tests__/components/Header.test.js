import React from 'react'
import { shallow } from 'enzyme'

import { Header } from '@/components/blocks'

describe('Header', () => {
  test('match snapshot', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })
})
