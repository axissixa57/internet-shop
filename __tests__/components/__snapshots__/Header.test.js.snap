// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header match snapshot 1`] = `
<Styled(PageHeader)
  subTitle={
    <React.Fragment>
      <Memo(Connect(withRouter(SearchContainer))) />
      <Memo(Connect(UserbarContainer)) />
    </React.Fragment>
  }
  title={
    <NavLink
      to="/"
    >
      TestShop
    </NavLink>
  }
/>
`;
